torch/csrc/jit/type.h

```diff
@@ -31,7 +31,7 @@ struct Type;
 using TypePtr = std::shared_ptr<Type>;
 
 
-struct TORCH_API Type {
+struct TORCH_API Type : std::enable_shared_from_this<Type> {
```

Comment:
hm is there a way to do this without enable_shared_from_this? There's a lot of pitfalls using that and I'd rather avoid it. If you wanna go ahead with it, please add something that forbids creation of `Type` objects on the stack, i.e. make all constructors private and force people to use something like `make_shared` (or `make_type`). Otherwise people will create objects on the stack, try to get a `shared_ptr` from it, and bad things will happen.

Reply:
I think it is fine to use shared_from_this we used to have it on Type and never ran into any problems.